qualtran/cirq_interop/_cirq_to_bloq_test.py

```diff
@@ -77,16 +78,18 @@ def test_cirq_gate_as_bloq_for_trivial_gates():
     assert toffoli.signature[0].shape == (3,)
 
     assert str(x) == 'cirq.X'
-    assert str(rx) == 'cirq.Rx'
+    assert str(rx) == 'cirq.Rx(0.123π)'
     assert str(toffoli) == 'cirq.TOFFOLI'
 
 
 def test_cirq_gate_as_bloq_tensor_contract_for_and_gate():
     and_gate = And()
     bb = BloqBuilder()
     ctrl = [bb.add(OneState()) for _ in range(2)]
-    ctrl, target = bb.add(CirqGateAsBloq(and_gate), ctrl=ctrl)
-    cbloq = bb.finalize(ctrl=ctrl, target=target)
+    target = bb.add(ZeroState())
+    q = [*ctrl, target]
+    c0, c1, target = bb.add(CirqGateAsBloq(and_gate), q=q)
```

Comment:
This seems like a backwards incompatible change? The CirqGateAsBloq doesn't preserve the signature anymore if the underlying gate is a `GateWithRegisters` ?

In theory, this should be fine since `GateWithRegisters` are bloqs so we should use them without wrapping in `CirqGateAsBloq`, but I'm nervous that this might break stuff in the interop. The tests are passing so that's a good indication, but have you looked into any potential consequence of this?

Reply:
One way to be full proof could be to raise an error if a user tries to wrap a `GateWithRegisters` into a `CirqGateAsBloq` (or any other object that satisfies `isinstance(gate, Bloq)`). 

I don't like the current behavior where it silently works, but does the wrong thing. It's better to explicitly raise an error if we don't want / expect users to wrap bloq objects in this wrapper

Reply:
Yes, I can add some error checking since this changes the behavior of using `CirqGateAsBloq` with a `GateWithRegisters`. 

It was on purpose to make `CirqGateAsBloq` only use the `cirq.Gate` API, which doesn't have a notion of signatures. If we wanted to preserve the "special casing", I would (theoretically) argue for a different adapter: `GateWithRegistersAsBloq`... but this is redundant :)

Reply:
actually, I don't know if an error is appropriate: since `GateWithRegisters` is a `cirq.Gate`, you should be able to wrap it -- but it will treat it as a cirq gate. Maybe I'll emit a warning?

Reply:
It won't silently fail anyways: if you were wrapping a sided GateWithRegisters with CirqGateAsBloq, the change will change the signature; which will raise an error when you're trying to wire it up. 

And of course the fix is simple: remove the adapter! I've added a warning